node/mocks/cp_client.go

```diff
@@ -0,0 +1,300 @@
+// Copyright 2014 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package agent implements a service that runs on a serviced node. It is
+// responsible for ensuring that a particular node is running the correct services
+// and reporting the state and health of those services back to the master
+// serviced.
+
+// +build unit integration
+
+package mocks
+
+import (
+	"github.com/control-center/serviced/dao"
+	"github.com/control-center/serviced/domain"
+	"github.com/control-center/serviced/domain/addressassignment"
+	"github.com/control-center/serviced/domain/service"
+	"github.com/control-center/serviced/domain/servicestate"
+	"github.com/control-center/serviced/domain/servicetemplate"
+	"github.com/control-center/serviced/domain/user"
+	"github.com/control-center/serviced/metrics"
+	"github.com/control-center/serviced/volume"
+)
+
+type ControlClient struct {
```

Comment:
Technically speaking you have built a mock here, but it has very limited utility because none of behavior can be controlled for testing purposes.  Please try this tool - https://github.com/vektra/mockery. It will generate a true mock for you - one that can use this [mocking package](https://github.com/stretchr/testify#mock-package). 

From the perspective of the unit-test, everything should work the same. The advantage is that anyone else who tries to use this mock in the future will have a full-feature mock at their disposal.